src/basic/fileio.c

```diff
@@ -33,6 +33,8 @@
 
 /* The maximum size of the file we'll read in one go in read_full_file() (64M). */
 #define READ_FULL_BYTES_MAX (64U * U64_MB - UINT64_C(1))
+/* The maximum size of large files that we'll read in one go in read_full_file() (1G). */
+#define READ_FULL_LARGE_BYTES_MAX (1U * U64_GB - UINT64_C(1))
```

Comment:
This seems iffy. There are many systems where reading of 1GB of file into memory is not going to work.

Reply:
Should be done